_drafts/diving-into-jtag-part6.md

```diff
@@ -0,0 +1,373 @@
+to any instruction, and allows us to perform step-by-step execution of instructions. We
+can also read and write the contents of the processor's general purpose registers. To read
+from protected memory, we need to find a `load word` instruction with an operand register,
+set the target address in the operand register, and execute this single instruction. Since
+we can't read flash memory, we don't know where the instructions are located, so it may seem
+difficult to find the read word instruction. However, all we need is an instruction that
+reads memory from an address in some register to a register, which is a fairly common operation.
+
+This method is described in detail in the article [Firmware dumping technique for an ARM Cortex-M0 SoC](https://blog.includesecurity.com/2015/11/firmware-dumping-technique-for-an-arm-cortex-m0-soc/) I highly recommend reading it.
+
+### nRF52 Debug Resurrection (APPROTECT Bypass)
+
+Here the situation is more complicated and the security level prohibits access
+to Debug Port. I.e. the previous method will not work.
+
+This is where **voltage glitching** comes in. Voltage glitching (also known as
+fault injection or power glitching) is a technique used in hacker attacks on
+microcontrollers and other embedded systems to disrupt their normal operation
+and gain access to protected information or functions. This technique is based
+on temporarily changing (glitching) the power supply voltage of a device, causing
+it to malfunction. While the microcontroller is running, short bursts (glitches)
+of low or high voltage are applied at certain points in time that disrupt the normal
+operation of the device. These glitches can cause errors in the execution of
+processor instructions, such as skipping or incorrect execution of individual
+instructions. This can cause the microcontroller to miss important safety checks.
+If a glitch is successfully synchronized, it can disable security mechanisms or
+access protected memory, allowing an attacker to retrieve sensitive data or gain
+control of the system.
+
+The main goal for this method is to find the exact time interval to apply this glitch.
+
+The author of the article determines the exact time when the AHB-AP has to initialize
+itself according to the value of the APPROTECT, which is stored in UICR (a dedicated
+Flash memory partition) and applies the glitch at that exact time, which allows him
+to bypass the protection and gain access to the DebugPort.
+
+Again see the original article [nRF52 Debug Resurrection (APPROTECT Bypass) Part 1](https://limitedresults.com/2020/06/nrf52-debug-resurrection-approtect-bypass/)
+for more details.
+
+# Conclusion
+
+This article is a very superficial overview of the topic of embedded devices security
+in general and JTAG security in particular. This topic is very vast to cover in one
+article and unfortunately I am not an expert in this topic. :)
```

Comment:
Here is an example rephrase with a slightly more authoritative tone! This is a great
article but understand the sentiment of wanting to highlight that cybersecurity is not
your speciality:
```suggestion
This article has only scratched the surface of embedded devices security
in general and JTAG security in particular. If you have experience
in cyber security and/or JTAG attacks, I would love to hear what you have
learned and what topics I missed!
```